app/models/manageiq/providers/amazon/agent_coordinator.rb

```diff
@@ -137,10 +137,26 @@ def deploy_agent
     @deploying = true
 
     kp = find_or_create_keypair
-    zone_name = ec2.client.describe_availability_zones.availability_zones[0].zone_name
-    subnets = get_subnets(zone_name)
-    raise "No subnet_id is available for #{zone_name}!" if subnets.empty?
-    security_group_id = find_or_create_security_group(subnets[0].vpc_id)
+
+    vpc_ids = get_dns_enabled_vpc_ids
+    raise "Smartstate analysis needs a VPC whose enablDnsHostnames setting is true!" if vpc_ids.empty?
```

Comment:
Typo in "enablDnsHostnames" - should be "enableDnsHostnames"

Reply:
Thanks, I'll change it.